### PR TITLE
Move train_cfg/test_cfg into model

### PR DESCRIPTION
**Pull request by xvjiarui**

### CODE REVIEW

File: .dev_scripts/batch_test.py [hunk 1]
```diff
@@ -157,8 +157,8 @@ def main():
             shuffle=False)
 
         # build the model and load checkpoint
-        model = build_detector(
-            cfg.model, train_cfg=None, test_cfg=cfg.test_cfg)
+        cfg.model.train_cfg = None
+        model = build_detector(cfg.model, test_cfg=cfg.get('test_cfg'))
```
Comment (ZwwWayne) [inline .dev_scripts/batch_test.py:161]: Should test_cfg also been changed to be cfg.model.test_cfg?

**Comment by ZwwWayne** [comment]
LGTM, see if @hellock has any comments.

**Comment by xvjiarui** [comment]
Kindly ping @hellock

**Comment by ZwwWayne** [comment]
1. We also need to update documentation about configs, including the config examples.
2. Add deprecation notification in the documentation about configs.

**Review by hellock — APPROVED** [review]

**Comment by hhaAndroid** [comment]
@xvjiarui  I tested all models using batch_test.py without checkpoint and find missed modifying the following configuration：

1 configs/faster_rcnn/faster_rcnn_r50_fpn_soft_nms_1x_coco.py 
2 configs/faster_rcnn/faster_rcnn_r50_fpn_ohem_1x_coco.py
3 configs/lvis/mask_rcnn_r50_fpn_sample1e-3_mstrain_1x_lvis_v1.py
4 configs/lvis/mask_rcnn_r50_fpn_sample1e-3_mstrain_2x_lvis_v0.5.py

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**